oak/server/rust/oak_runtime/src/node/grpc/server/mod.rs

```diff
@@ -504,64 +521,126 @@ impl GrpcRequestHandler {
                 err
             );
         }
-        if let Err(err) = self.runtime.channel_close(response_writer) {
-            error!(
-                "Failed to close response writer channel for invocation: {:?}",
-                err
-            );
+
+        Ok(GrpcResponseIterator::new(
+            self.runtime.clone(),
+            response_reader,
+            metrics_data,
+            self.method_name.clone(),
+        ))
+    }
+}
+
+struct GrpcResponseIterator {
+    runtime: RuntimeProxy,
+    response_reader: oak_abi::Handle,
+    metrics_data: Metrics,
+    method_name: String,
+    _timer: prometheus::HistogramTimer,
```

Comment:
Why is it necessary?

Reply:
It records the time taken to process the responses; when the timer is dropped it records the duration. 

Added a comment.